common/common/src/main/java/io/helidon/common/LruCacheImpl.java

```diff
@@ -47,32 +47,31 @@ final class LruCacheImpl<K, V> implements LruCache<K, V> {
     public Optional<V> get(K key) {
         readLock.lock();
 
-        V value;
         try {
-            value = backingMap.get(key);
+            V value = backingMap.get(key);
+            if (null == value) {
```

Comment:
Moving this null check into the try/finally block will just make us hold the `readLock` a bit longer. Can you please explain the reason for moving it here?